wgsl/index.bs

```diff
@@ -4888,13 +4898,74 @@ name of a variable.  See [[#forming-references-and-pointers]] for other cases.
       </xmp>
     </div>
 
-<pre class='def'>
-assignment_statement
-  : unary_expression EQUAL short_circuit_or_expression
-      If unary_expression is a variable, this maps to OpStore to the variable.
-      Otherwise, unary expression is a pointer expression in an Assigning (L-value) context
-      which maps to OpAccessChain followed by OpStore
-</pre>
+### Phony Assignment ### {#phony-assignment-section}
+
+When the [=left-hand side=] of an assignment is an underscore token,
+the assignment is an <dfn noexport>phony assignment</dfn>:
+The right-hand side is evaluated, and then ignored.
+
+<table class='data'>
+  <thead>
+    <tr><th>Precondition<th>Statement<th>Description
+  </thead>
+  <tr algorithm="phony-assignment">
+    <td>|e|: |T|,<br>
+        |T| is [=constructible=], or a [=pointer type=]
+    <td class="nowrap">_ = |e|
+    <td>Evaluates |e|.
+
+        Note: The resulting value is not stored.
+        The `_` token is not an identifier, and therefore cannot be used in an expression.
+</table>
+
+A phony-assignment is useful for:
+* Calling a function that returns a value, but clearly expressing that the resulting value
+    is not needed.
+* [=statically accessed|Statically accessing=] a variable, thus establishing it as a part of
+    the [=resource interface of a shader|shader's resource interface=].
+
+    Note: A variable's store type may not be constructible, e.g. it contains an atomic type, or a runtime-sized array.
+    In these cases, use a pointer to the variable's contents instead.
+
+<div class='example wgsl global-scope' heading='Using phony-assignment to throw away an un-needed function result'>
+  <xmp highlight=rust>
+    var<private> counter: i32;
+
+    fn increment_and_yield_previous() -> i32 {
+      let previous = counter;
+      counter = counter + 1;
+      return previous;
+    }
+
+    fn user() {
+      // Increment the counter, but don't use the result.
+      _ = increment_and_yield_previous();
+    }
+  </xmp>
+</div>
+
+
+<div class='example wgsl global-scope' heading='Using phony-assignment to occupy bindings without using them'>
+  <xmp highlight=rust>
+    [[block]] struct BufferContents {
+        counter: atomic<u32>;
+        data: array<vec4<f32>>;
+    };
+    [[group(0),binding(0)]] var<storage> buf: BufferContents;
+    [[group(0),binding(1)]] var t: texture_2d<f32>;
+    [[group(0),binding(2)]] var s: sampler;
+
+    [[stage(fragment)]]
+    fn shade_it() -> [[location(0)]] vec4<f32> {
+      // Declare that buf, t, and s are part of the shader interface, without
+      // using them for anything.
+      _ = &buf;
+      _ = &t;
```

Comment:
any reason this isn't assigning by value?

Reply:
textures and samplers are not constructible.

The constraint on the RHS of phony-assignment is constructible or pointer, to make it as close as possible to regular assignment,  which only allows constructible.

There's a special exception for passing textures and samplers as function call parameters.

Reply:
But [we don't support the address-of operator on textures/samplers](https://github.com/gpuweb/gpuweb/pull/1983), so this example code is not valid. If we want to use this phony-assignment to retain sampler/texture bindings we'd need it to support assigning textures and samplers too.

Reply:
ok, so I guess we'll have to add those cases too.  :-)